### ISSUE TITLE
AnimatedSprite: current texture index does not match current frame index

### ISSUE THREAD

**Issue by FNeves88** [pixijs/pixi.js#6465]
I have an AnimatedSprite and, let's say, 95% of the time on the first run it doesnt work properly. In this example, it's a toggle button, first image is frame 0, second image is last frame, and third image is what it likes when I run it on the first time.

![button_animatedsprite](https://user-images.githubusercontent.com/1436559/76209289-6bb50d80-61f9-11ea-9a78-dbc4f41a0367.png)

The frame where it stops it's not always the same, could be last but one or in the middle..
It also occurs after first run but it is very very rare, most of the time is on first run while my webpage is still building things.

I printed some of the internal counters and I noticed that onFrameChange does not print for all frames, and the onComplete method always fires and currentFrame matches with the last index:

![Screenshot 2020-03-06 at 09 42 28](https://user-images.githubusercontent.com/1436559/76209794-a66b7580-61fa-11ea-8c04-9da4e63b0487.png)

When I inspect the object, it says that is in last frame (index 10):

![Screenshot 2020-03-06 at 09 43 33](https://user-images.githubusercontent.com/1436559/76209889-ddda2200-61fa-11ea-8cc6-67c235e5f23c.png)

But then, the current texture is the relative to frame index 9, as you can see in the textureCacheIds:

![Screenshot 2020-03-06 at 09 46 58](https://user-images.githubusercontent.com/1436559/76209904-ec283e00-61fa-11ea-985a-b5b958d7e550.png)

And here are the list of all textures:

![Screenshot 2020-03-06 at 09 46 31](https://user-images.githubusercontent.com/1436559/76210196-88eadb80-61fb-11ea-814c-3762b9605fb6.png)

So, it seems that internally AnimatedSprite is correct, but visually it's not.
I noticed this since I updated from v5.2.0 to v5.2.1, I don't know if it is related to #6274 

### Environment

- **`pixi.js` version**: _5.2.1_
- **Browser & Version**: _Chrome 80.0.3987.132_
- **OS & Version**: _macOS Catalina_
- **Dev Environment**: _Webpack 4.42.0_

**Comment by ivanpopelyshev** [comment]
Hello! The thing with AnimatedSprite - it is a baseline. Its not something that people actually use in production, most of pixi users patch it. It is also not something that requires intimate knowledge of webgl.

https://github.com/pixijs/pixi.js/blob/dev/packages/sprite-animated/src/AnimatedSprite.ts

There can be two type of problems:
1. Problem is in AnimatedSprite
2. Problem is in internal `_texture` assignment in Sprite.

There are two ways you can speed up resolution:
1. Find the problem yourself, patch it, make PR
2. Reproduce it in a demo and someone will look at it soon.

Alternatively:
1. Just wait and maybe someone does it for you.
2. Wait when a specialist of AnimatedSprite looks at this thread and suddently remembers that he forgot to patch this particular bug.

**Comment by FNeves88** [comment]
I guess I found out the problem:

In the `PIXI.AnimatedSprite.prototype.update` we have an `if/else if/else if` at the end, and only for the last `else if` the method `this.updateTexture` is called. If code enters in the first two, this is never called and if the last drawn frame was one in the middle, the animation will appear with that frame visible instead of the last one.

I checked commit log in #6274 and this was changed. Before, in the first two _ifs_ the method `gotoAndStop` was called, wich contains a call to `updateTexture`, and now it was substituted by a simple call to `stop()` and never update the texture in that cases.

![Screenshot 2020-03-09 at 17 35 22](https://user-images.githubusercontent.com/1436559/76241113-61f8cd80-622c-11ea-8259-7d43867b8b29.png)

And I guess that this situation occurs more often when we have some kind of frame skipping/lag.
